trings/UI/Views/WalkthroughGuide/WalkthroughGuideViewController.swift

```diff
@@ -9,65 +9,64 @@ import UIKit
 
 class WalkthroughGuideViewController: UIViewController {
 
-    @IBOutlet private weak var guideScrollView: UIScrollView! {
+    @IBOutlet private weak var doneButton: UIButton! {
         didSet {
-            guideScrollView.delegate = self
+            doneButton.setTitle(L10n.Walkthrough.ContinueButton.title, for: .normal)
         }
     }
+    
     @IBOutlet private weak var guidePageControl: UIPageControl! {
         didSet {
-            guidePageControl.numberOfPages = 0
             guidePageControl.currentPageIndicatorTintColor = Asset.Colors.primaryGreen.color
             guidePageControl.pageIndicatorTintColor = Asset.Colors.grayLight.color
         }
     }
-    @IBOutlet private weak var doneButton: UIButton! {
-        didSet {
-            doneButton.setTitle(L10n.Walkthrough.ContinueButton.title, for: .normal)
-        }
-    }
-
-    @IBOutlet private weak var onTheGroundTextLabel: UILabel! {
-        didSet {
-            onTheGroundTextLabel.numberOfLines = 0
-        }
-    }
-    @IBOutlet private weak var onTheGroundTitle: UILabel! {
-        didSet {
-            onTheGroundTitle.numberOfLines = 0
-        }
-    }
-    @IBOutlet private weak var onTheGroundImage: UIImageView!
-
-    @IBOutlet private weak var inTheCloudTextLabel: UILabel! {
-        didSet {
-            inTheCloudTextLabel.numberOfLines = 0
-        }
-    }
-    @IBOutlet private weak var inTheCloudTitle: UILabel! {
-        didSet {
-            inTheCloudTitle.numberOfLines = 0
-        }
-    }
-    @IBOutlet private weak var inTheCloudImage: UIImageView!
-
-    @IBOutlet private weak var inYourWalletTextLabel: UILabel! {
-        didSet {
-            inYourWalletTextLabel.numberOfLines = 0
-        }
-    }
-    @IBOutlet private weak var inYourWalletTitle: UILabel! {
-        didSet {
-            inYourWalletTitle.numberOfLines = 0
-        }
-    }
-    @IBOutlet private weak var inYourWalletImage: UIImageView!
-
+    
+    @IBOutlet private weak var pageContainerView: UIView!
+    
+    private var pageViewController: UIPageViewController = {
+        let pageViewController = UIPageViewController(transitionStyle: .scroll, navigationOrientation: .horizontal)
+        pageViewController.view.translatesAutoresizingMaskIntoConstraints = false
+        return pageViewController
+    }()
+    
     var viewModel: WalkthroughGuideViewModel?
+    
+    private var pendingIndex = 0
+    private let walkthroughPages: [WalkthroughPage] = [
```

Comment:
I feel like here the view controller is assuming we have 3 pages, but really it should be able to handle as many pages as the view model gives it.

Reply:
Ah makes sense, I will dynamically create the view controllers based on the numbers of pages from the view model